brave/brave-core
# Add IKEv2 support for status tray icon

Resolves https://github.com/brave/brave-browser/issues/32143

- Added IKEv2 support for status tray icon
- Status tray icon launched if any of BraveVPN protocols connected
- Show system notifications if brave IKEv2 vpn connected (only if dns helper started)

https://github.com/brave/brave-core/assets/2965009/141fa44a-85bc-4125-aee5-d3e82fe7c6b6




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Check that Brave VPN status tray icon allow to manage both BraveVPN protocols, Ras/Wireguard
- Check if users enable vpn using system Brave VPN RAS connection entry, the tray icons shows and updates connection status correctly
- Check that the icon becomes visible if it was not visible and IKEv2 connected.
- Check that browser shows system notifications if Brave IKEv2 vpn connected (only if dns helper started)
